### DISCUSSION TITLE
mp3-http-stream example works on esp32 but not on esp32s2-wrover OR lolin_s2mini

### DISCUSSION THREAD

**Discussion by ptunstall** [Moddable-OpenSource/moddable#1233] [Q&A] [answered]
Wifi connection and mp3 steaming work perfectly on an esp32-wroom via:
```
mcconfig -d -m -p esp32 ssid="SSID_NAME" password="password"
```

But the same command ran on both the esp32-s2-wrover and the lolin_s2mini never even connect to wifi.

Anyone experience this?

**Comment by phoddie** [comment]
Cool that the MP3 streaming is working for you. That's a powerful capability.

The two boards you mention use the ESP32-S2 chip. When you build with `-p esp32` you are building for the original ESP32 chip. You need to indicate which board to build for. For the two boards you mention that would be:

```
mcconfig -d -m -p esp32/lolin_s2mini ssid="SSID" password="PASSWORD"
mcconfig -d -m -p esp32/saola_wrover ssid="SSID" password="PASSWORD"
```

Our ESP32 documentation [lists supported boards](https://github.com/Moddable-OpenSource/moddable/blob/public/documentation/devices/esp32.md#platforms) along with the platform/sub-platform ID to use when building.

**Reply by ptunstall** [reply]
Oh yes, Apologies I was changing to the correct target device.  It uploads fine, just xsbug says "lost connection" where as the regular esp32 connects and begins playing.

**Reply by phoddie** [reply]
Thanks for the clarification. We'll take a look.

**Comment by mkellner** [comment]
What audio output device are you using and how is it connected?

I am able to reproduce the problems you're having with the lolin_s2mini and saola_wrover. I'll have a fix for it soon.

**Reply by ptunstall** [reply]
Hey! I am using i2s via the `max98357A` i2s amplifier.  There's just a simple 3w 8ohm speaker on the other end of this i2s amp.

**Comment by mkellner** [comment, marked as answer]
For the saola_wrover, try replacing `$MODDABLE/build/devices/esp32/sdkconfig-s2-psram/sdkconfig.defaults` with the contents of this gist:

https://gist.github.com/mkellner/b244fa68827cef830236f956b09db8c6

My max98357 is connected to pins 1, 2 and 3.

In the defines section of `$MODDABLE/build/devices/esp32/targets/saola_wrover/manifest.json`, I have:

```
    "defines": {
        "audioOut": {
            "bitsPerSample": 16,
            "numChannels": 1,
            "volume_divider": 2,
            "i2s": {
                "bitsPerSample": 16,
                "bck_pin": 2,
                "lr_pin": 3,
                "dataout_pin": 1
            }
        }
    }
```

With that, I can run mp3-http-stream successfully.

Let me know how it works for you.

https://github.com/Moddable-OpenSource/moddable/assets/6822704/6232b433-2433-44a3-b254-0b6a77284951

**Reply by ptunstall** [reply]
This works perfectly on the saola_wrover! the s2mini seems to work correctly too with the output in xsbug but I can't get audio out on pins 3,5,7 but thats most likely an error on my end.  Going to solder the rest of the pins in a bit so I can use 1,2,3 like you're doing on the saola.

Can also conform that the elevenlabs example also works WHICH IS FLIPPIN NEATO!!!!

Really appreciate you guys so much for this!!!

**Reply by mkellner** [reply]
For the `lolin_s2mini`, change this file:

`$MODDABLE/build/devices/esp32/targets/lolin_s2mini/sdkconfig/sdkconfig.defaults`

Change the line:

```
CONFIG_SPIRAM_USE_CAPS_ALLOC=y
```

to

```
CONFIG_SPIRAM_USE_MALLOC=y
```

If you have problems try building clean first.

```
mcconfig -d -m -p esp32/lolin_s2mini ... -t clean
```

**Reply by ptunstall** [reply]
YES! This works perfectly for the `lolin s2mini`!!! Wizards!  THANK YOU!!
